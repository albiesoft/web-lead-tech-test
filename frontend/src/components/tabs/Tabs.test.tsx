import { render, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import Tabs, { type TabsProps } from "./Tabs";

describe("<Tabs>", () => {
  const mockOnChange = vi.fn();
  const renderComponent = (props?: TabsProps<string>) => {
    const defaultProps: TabsProps<string> = {
      tabs: [
        {
          id: "tab1",
          label: "Tab 1",
        },
        {
          id: "tab2",
          label: "Tab 2",
        },
        {
          id: "tab3",
          label: "Tab 3",
        },
      ],
      activeTab: "tab1",
      onTabChange: mockOnChange,
    };
    return render(<Tabs {...(!!props ? props : defaultProps)} />);
  };

  it("should match the snapshot", () => {
    const comp = renderComponent();
    expect(comp).toMatchSnapshot();
  });

  it("should trigger onTabChange when click", () => {
    const comp = renderComponent();

    fireEvent.click(comp.getByTestId("tab-tab2"));

    expect(mockOnChange).toHaveBeenCalled();
  });
});
