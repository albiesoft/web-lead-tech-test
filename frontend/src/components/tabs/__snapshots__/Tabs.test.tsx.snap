// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Tabs> > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex shadow-[0_10px_10px_rgba(0,0,0,0.05)]"
      >
        <li
          class="flex-1 text-center cursor-pointer font-semibold border-b-2 border-transparent py-1 transition-all duration-200 ease-in-out text-accent border-accent"
          data-testid="tab-tab1"
        >
          Tab 1
        </li>
        <li
          class="flex-1 text-center cursor-pointer font-semibold border-b-2 border-transparent py-1 transition-all duration-200 ease-in-out "
          data-testid="tab-tab2"
        >
          Tab 2
        </li>
        <li
          class="flex-1 text-center cursor-pointer font-semibold border-b-2 border-transparent py-1 transition-all duration-200 ease-in-out "
          data-testid="tab-tab3"
        >
          Tab 3
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex shadow-[0_10px_10px_rgba(0,0,0,0.05)]"
    >
      <li
        class="flex-1 text-center cursor-pointer font-semibold border-b-2 border-transparent py-1 transition-all duration-200 ease-in-out text-accent border-accent"
        data-testid="tab-tab1"
      >
        Tab 1
      </li>
      <li
        class="flex-1 text-center cursor-pointer font-semibold border-b-2 border-transparent py-1 transition-all duration-200 ease-in-out "
        data-testid="tab-tab2"
      >
        Tab 2
      </li>
      <li
        class="flex-1 text-center cursor-pointer font-semibold border-b-2 border-transparent py-1 transition-all duration-200 ease-in-out "
        data-testid="tab-tab3"
      >
        Tab 3
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
